server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/LongValuesComparatorSource.java

```diff
@@ -120,7 +120,7 @@ public FieldComparator<?> newComparator(String fieldname, int numHits, boolean e
         final long lMissingValue = (Long) missingObject(missingValue, reversed);
         // NOTE: it's important to pass null as a missing value in the constructor so that
         // the comparator doesn't check docsWithField since we replace missing values in select()
-        return new LongComparator(numHits, fieldname, null, reversed, enableSkipping && this.enableSkipping) {
+        return new LongComparator(numHits, fieldname, lMissingValue, reversed, enableSkipping && this.enableSkipping) {
```

Comment:
The change of missingValue argument seems to be unrelated to the subject of this pull request, I feel strongly we should move it off to separate pull request

Reply:
Sure, let me remove this.

Reply:
Cool, I willl raise PR for missing value once this is merged

Reply:
cc @backslasht @kkmr -> I will be sending new PR for last commit as suggested by reta.
2a4a43009bab047d59aeff65e4166c57bf1796e9